packages/protocol/test/governance/vesting.ts

```diff
@@ -0,0 +1,1734 @@
+      await createNewVestingInstanceTx(vestingDefaultSchedule, web3)
+      const vestingInstanceRegistryAddress = await vestingFactoryInstance.vestings(beneficiary)
+      const vestingInstance = await VestingInstance.at(vestingInstanceRegistryAddress)
+      await assertRevert(vestingInstance.revoke({ from: accounts[5] }))
+    })
+
+    it('should revert if vesting is already revoked', async () => {
+      await createNewVestingInstanceTx(vestingDefaultSchedule, web3)
+      const vestingInstanceRegistryAddress = await vestingFactoryInstance.vestings(beneficiary)
+      const vestingInstance = await VestingInstance.at(vestingInstanceRegistryAddress)
+      await vestingInstance.revoke({ from: revoker })
+      await assertRevert(vestingInstance.revoke({ from: revoker }))
+    })
+
+    it('should revert if vesting is non-revokable', async () => {
+      const vestingSchedule = _.clone(vestingDefaultSchedule)
+      vestingSchedule.vestingRevocable = false
+      await createNewVestingInstanceTx(vestingSchedule, web3)
+      const vestingInstanceRegistryAddress = await vestingFactoryInstance.vestings(beneficiary)
+      const vestingInstance = await VestingInstance.at(vestingInstanceRegistryAddress)
+      await assertRevert(vestingInstance.revoke({ from: revoker }))
+    })
+
+    it('should revert if vesting is paused', async () => {
+      await createNewVestingInstanceTx(vestingDefaultSchedule, web3)
+      const vestingInstanceRegistryAddress = await vestingFactoryInstance.vestings(beneficiary)
+      const vestingInstance = await VestingInstance.at(vestingInstanceRegistryAddress)
+      await assertRevert(vestingInstance.pause(366 * DAY, { from: revoker }))
+      await assertRevert(vestingInstance.revoke({ from: revoker }))
+    })
+  })
+
+  describe('#getInitialVestingAmount()', () => {
+    it('# should always return the initial vesting amount at the time of construction', async () => {
+      await createNewVestingInstanceTx(vestingDefaultSchedule, web3)
+      const vestingInstanceRegistryAddress = await vestingFactoryInstance.vestings(beneficiary)
+      const vestingInstance = await VestingInstance.at(vestingInstanceRegistryAddress)
+      const initialVestingAmount = await vestingInstance.getInitialVestingAmount()
+      assertEqualBN(
+        initialVestingAmount,
+        vestingDefaultSchedule.vestAmountPerPeriod.multipliedBy(
+          vestingDefaultSchedule.vestingNumPeriods
+        )
+      )
+    })
+  })
+
+  describe('#refundAndFinalize()', () => {
+    let vestingInstanceRegistryAddress
+    let vestingInstance
+
+    beforeEach(async () => {
+      await createNewVestingInstanceTx(vestingDefaultSchedule, web3)
+      vestingInstanceRegistryAddress = await vestingFactoryInstance.vestings(beneficiary)
+      vestingInstance = await VestingInstance.at(vestingInstanceRegistryAddress)
+      // wait some time for some gold to vest
+      const timeToTravel = 7 * MONTH
+      await timeTravel(timeToTravel, web3)
```

Comment:
Can we make the test more complicated by having the beneficiary withdraw some gold first?

Reply:
added